### ISSUE TITLE
Simulated data updates for WGSAM skill assessment

### ISSUE THREAD

**Issue by sgaichas** [NOAA-EDAB/ms-keyrun#56]
After dataset demonstration 2 February 2023 we developed a list of requests for additions. @vbartolino please check to see what is missing:

- [ ] Add datasets with subannual time steps (current interval is 73 days, 5x per year) including a column of time steps for `simFishery` datasets 
  - [ ] Index
  - [ ] Age structure
  - [ ] Length structure
  - [ ] Weight at age
- [ ] Add time steps of year for surveys to `simSurvey` datasets
- [ ] Add table describing agecl and age for each species
- [ ] Add parameters to `simBiolPar`
  - [ ] Maturity ogive
  - [ ] Gape width/or derived prey selection parameters

**Comment by slucey** [comment]
@sgaichas do you want to add biomass data for non-focal species to the list?

**Comment by sgaichas** [comment]
Yes, thank you Sean, added. Did I miss any?

**Comment by vbartolino** [comment]
If possible also catches in `simCatchIndex` should be provided by timestep

**Comment by sgaichas** [comment]
@vbartolino yes, I will clarify that. Do the fishery age, length, and weight at age also need to be for each timestep?

**Comment by slucey** [comment]
> 

Looks good ... thanks!

**Comment by slucey** [comment]
> Yes, thank you Sean, added. Did I miss any?

Looks good to me.

**Comment by vbartolino** [comment]
> @vbartolino yes, I will clarify that. Do the fishery age, length, and weight at age also need to be for each timestep?

I do not think it is not strictly required but for well monitored stocks usually we have sampling programmes which try to collect samples in each quarter. I suggest to retrieve those quantities from all the quarters as a routine and eventually challenge the model(s) later on providing only a subset of timesteps

**Comment by grantdadams** [comment]
Thanks for all the work on this! It was really straightforward for me to grab what I needed for CEATTLE. I had a few follow up questions and requests apart from the above. Maybe some will be left for individual modelers to decide:

- [ ] In what months does the spring and fall survey occur?
- [ ] For each species, are selectivity and catchability the same for spring and fall surveys?
- [ ] What is the shape of selectivity for each species/fleet (dome or asymptotic)? 
- [ ] What weight index to use for population level biomass and spawning biomass for each species?
- [ ] What month do each species spawn?

Data requests:
- [ ] simFisheryAgeLencomp data to make age to length conversion matrices for the fishery length comp data. Similar to the simSurveyAgeLencomp. The ages ranges from the survey and fishery don't overlap for some species (otherwise I'd use the data from the survey). But also happy to make some assumptions here if its a lot of work!

**Comment by sgaichas** [comment]
Sorry for delays all; I'm beginning to address these requests on branch [sarah_wgsamsim](https://github.com/NOAA-EDAB/ms-keyrun/tree/sarah_wgsamsim) and will check them off as they are done. When finished we will pull them back into the main branch

**Comment by grantdadams** [comment]
Where can we find the table for converting agecl to age?

**Comment by sgaichas** [comment]
@grantdadams these have been added to the simBiolPar dataset, currently only in the [sarah_wgsamsim](https://github.com/NOAA-EDAB/ms-keyrun/tree/sarah_wgsamsim) branch. description [here](https://github.com/NOAA-EDAB/ms-keyrun/blob/sarah_wgsamsim/man/simBiolPar.Rd)

**Comment by gustavdelius** [comment]
- [ ] Maturity ogive as a function of length or weight instead of age.

@sgaichas has added the maturity as a function of ageclass to the `simBiolPar` data frame. I believe it is common to observe maturity ogives as a function of length or weight and this would be more useful for size-based models. Can this be extracted easily from Atlantis?

**Comment by gustavdelius** [comment]
- [ ]  a simulated data frame `simSurveyLenDietcomp` similar to `surveyLenDietcomp`, containing the lengths or weights of predator and prey.

**Comment by gustavdelius** [comment]
Is any information available about how the fishing effort has changed over time?

**Comment by vbartolino** [comment]
> a simulated data frame simSurveyLenDietcomp similar to surveyLenDietcomp, containing the lengths or weights of predator and prey.

this would be relevant also for a model like gadget where data diet composition can be specified by predator and preys size groups
